graphqlapi/src/main/java/io/stargate/graphql/core/DataFetchers.java

```diff
@@ -0,0 +1,452 @@
+import com.datastax.oss.driver.api.querybuilder.update.Update;
+import com.datastax.oss.driver.api.querybuilder.update.UpdateStart;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import graphql.schema.SelectedField;
+import io.stargate.auth.AuthenticationService;
+import io.stargate.auth.StoredCredentials;
+import io.stargate.auth.UnauthorizedException;
+import io.stargate.db.ClientState;
+import io.stargate.db.Persistence;
+import io.stargate.db.QueryState;
+import io.stargate.db.datastore.DataStore;
+import io.stargate.db.datastore.ResultSet;
+import io.stargate.db.datastore.Row;
+import io.stargate.db.datastore.schema.Column;
+import io.stargate.db.datastore.schema.Keyspace;
+import io.stargate.db.datastore.schema.Table;
+import io.stargate.graphql.graphqlservlet.HTTPAwareContextImpl;
+
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.literal;
+
+public class DataFetchers {
+    private final NameMapping nameMapping;
+    private final Persistence persistence;
+    private AuthenticationService authenticationService;
+
+    public DataFetchers(Persistence persistence, Keyspace keyspace, NameMapping nameMapping, AuthenticationService authenticationService) {
+        this.persistence = persistence;
+        this.authenticationService = authenticationService;
+
+        this.nameMapping = nameMapping;
+    }
+
+    abstract class AbstractMutationDataFetcher implements DataFetcher {
+        protected final Table table;
+        public AbstractMutationDataFetcher(Table table) {
+            this.table = table;
+        }
+
+        @Override
+        public Object get(DataFetchingEnvironment environment) throws Exception {
+            HTTPAwareContextImpl httpAwareContext = environment.getContext();
+//            if (httpAwareContext.getUserOrRole() != null) {
+//                statement.setOutgoingPayload(ImmutableMap.of("ProxyExecute", ByteBuffer.wrap(httpAwareContext.getUserOrRole().getBytes())));
+//            }
+
+            String token = httpAwareContext.getAuthToken();
+            StoredCredentials storedCredentials = authenticationService.validateToken(token);
+            ClientState clientState = persistence.newClientState(storedCredentials.getRoleName());
+            QueryState queryState = persistence.newQueryState(clientState);
+            DataStore dataStore = persistence.newDataStore(queryState, null);
+
+            String statement = buildStatement(table, environment, dataStore);
+            CompletableFuture<ResultSet> rs = dataStore.query(statement);
+            ResultSet resultSet = rs.get();
+
+            ImmutableMap.Builder resultMap = ImmutableMap.builder();
+            resultMap.put("value", environment.getArgument("value"));
+
+            return resultMap.build();
```

Comment:
Per my other comment on making data fetchers async, this is how we would transform this:
```suggestion
            return dataStore.query(statement)
                .thenApply(resultSet -> ImmutableMap.of("value", environment.getArgument("value")));
```